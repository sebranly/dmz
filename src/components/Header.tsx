import classnames from 'classnames';
import * as React from 'react';
import { getAnchorLink } from '../utils';

export interface HeaderProps {
  className?: string;
  text: string;
}

const Header: React.FC<HeaderProps> = (props) => {
  const { className, text } = props;
  const classes = classnames('font-bold my-5 py-1 text-xl text-lime-500', className);
  const anchorLink = getAnchorLink(text);

  return (
    <h3 className={classes} id={anchorLink}>
      <a href={`#${anchorLink}`}>{text}</a>
    </h3>
  );
};

export { Header };
