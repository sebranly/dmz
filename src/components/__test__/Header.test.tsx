import * as React from 'react';
import { render } from '@testing-library/react';
import { Header } from '../Header';

test('Header', () => {
  const createComponent = () => <Header text="Bonjour" />;

  const { container } = render(createComponent());
  expect(container.childNodes).toMatchSnapshot();
});
