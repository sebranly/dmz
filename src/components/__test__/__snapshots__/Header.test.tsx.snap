// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header 1`] = `
NodeList [
  <h3
    class="font-bold my-5 py-1 text-xl text-lime-500"
    id="bonjour"
  >
    <a
      href="#bonjour"
    >
      Bonjour
    </a>
  </h3>,
]
`;
